lib/SILOptimizer/Transforms/CSE.cpp

```diff
@@ -859,6 +923,152 @@ static bool isLazyPropertyGetter(ApplyInst *ai) {
   return false;
 }
 
+static void
+collectGuaranteedTransitiveUses(BeginBorrowInst *inst,
+                                SmallVectorImpl<SILInstruction *> &result) {
```

Comment:
@gottesmm Need your feedback on this. Do you think it'll be useful to move this to BorrowedOperand/Value ?